File: cyclestreets.app/src/main/java/net/cyclestreets/CycleStreets.java
package net.cyclestreets;

import android.os.Bundle;

import net.cyclestreets.addphoto.AddPhotoFragment;

public class CycleStreets extends MainNavDrawerActivity implements RouteMapActivity {
  @Override
  public void onCreate(final Bundle savedInstanceState) {
    MainSupport.switchMapFile(getIntent());

    super.onCreate(savedInstanceState);

    MainSupport.loadRoute(getIntent(), this);
  }

  @Override
  protected void onFirstRun() {
    Welcome.welcome(this);
  }

  @Override
  protected void onNewVersion() {
    Welcome.whatsNew(this);
  }

  @Override
  public void showMap() {
    showPage(0);
  }

  @Override
  protected void addDrawerItems() {
    addDrawerFragment(R.string.route_map,
                      R.drawable.ic_menu_mapmode,
                      RouteMapFragment.class);
    addDrawerFragment(R.string.itinerary,
                      R.drawable.ic_menu_agenda,
                      ItineraryAndElevationFragment.class,
                      new RouteAvailablePageStatus());
    addDrawerFragment(R.string.photomap,
                      R.drawable.ic_menu_gallery,
                      PhotoMapFragment.class);
    addDrawerFragment(R.string.photo_upload,
                      R.drawable.ic_menu_camera,
                      AddPhotoFragment.class);
    addDrawerFragment(BlogFragment.blogTitle(this),
                      -1,
                      BlogFragment.class);
    addDrawerFragment(R.string.settings,
                      android.R.drawable.ic_menu_preferences,
                      SettingsFragment.class);
  }
}

File: cyclestreets.app/src/main/java/net/cyclestreets/CycleStreets.kt
package net.cyclestreets

import android.content.Intent
import android.os.Bundle

class CycleStreets : MainNavDrawerActivity(), RouteMapActivity, PhotoMapActivity {
    public override fun onCreate(savedInstanceState: Bundle?) {
        MainSupport.switchMapFile(intent, this)

        super.onCreate(savedInstanceState)

        MainSupport.handleLaunchIntent(intent, this)
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        intent?.let {
            MainSupport.handleLaunchIntent(it, this)
        }
    }

    override fun onFirstRun() {
        Welcome.welcome(this)
    }

    override fun onNewVersion() {
        Welcome.whatsNew(this)
    }

    override fun showRouteMap() {
        showPage(R.id.nav_journey_planner)
    }

    override fun showPhotoMap() {
        showPage(R.id.nav_photomap)
    }
}
